src/type.ts

```diff
@@ -1,9 +1,17 @@
 import * as descriptor from "./compiler/descriptor.js";
```

Comment:
type registry shouldn't be getting this much involved in what options have been passed. I know that when you strip out the root package name it is really hard to find out what part of type_name is the package but I still feel like we could be move clever about this.

perhaps we could keep track of things twice with and without their package names to locate them quicker?  for instance, leading dot is a good clue to find out whether a symbol comes from the same file or from a dependency.  I'll think about this one.

Reply:
Totally understand what you're saying; how we pass the options feels sort of like an escape hatch.

But at the end of the day, both `getTypeReference` and `getTypeReferenceExpr` need to know if we are using namespaces or not. Let me know what your thoughts are!

Reply:
yep. I see your point. perhaps it is time to introduce something like compilation context using context pattern but I'll follow up with another PR.

Reply:
also, packageName is kind of confusing perhaps we should rename it to parentName.

Reply:
Changed!